src/pke/lib/scheme/bfvrns/bfvrns-leveledshe.cpp

```diff
@@ -806,26 +837,26 @@ Ciphertext<DCRTPoly> LeveledSHEBFVRNS::EvalFastRotation(ConstCiphertext<DCRTPoly
 
     const auto cryptoParams = std::dynamic_pointer_cast<CryptoParametersBFVRNS>(ciphertext->GetCryptoParameters());
 
-    /* In the HPSOVERQLEVELED mode, we do modulus switching to a smaller modulus before we start key switching.
-    The modulus switching was already done when computing the ciphertext digits using EvalFastRotationPrecompute.
-    The goal of the "if branch" below is to extract the current modulus Ql from the element parameters of one of
-    the digit polynomials (by removing the auxiliary moduli added for hybrid key switching).
-    ATTN: elemParams should not be a shared_ptr because it would modify digits. */
+    // We remove all auxiliary moduli P_i in the case of hybrid key switching
```

Comment:
Probably worth it to have a more detailed comment if EvalFastRotationPrecomp() drops some levels for one case.

Reply:
Added